### PR TITLE
Feature/157 store did

### PR DESCRIPTION
**Pull request by sadiabbasi**

### CODE REVIEW

File: test-integration/util/user.ts [hunk 1]
```diff
@@ -32,13 +32,15 @@ export async function createRandomID() {
 
     await user.createPerson();
     await user.updateKeys(password);
+    const chainID = randomString(8);
```
Comment (theblockstalk) [inline test-integration/util/user.ts:35]: this is not right, chainId comes from the getInfo endpoint on the blockchain API

File: test-integration/user.test.ts [hunk 2]
```diff
@@ -255,4 +255,16 @@ describe('User class', () => {
             await user.logout();
         })
     );
+    test(
+        'getDid() expect chainId and account name defined',
```
Comment (theblockstalk) [inline test-integration/user.test.ts:259]: there is no call to user.getDid()

**Review by theblockstalk — COMMENTED** [review]
there is no new function `getDid()` on the user class. this needs to be added, you can copy most of the code from 
https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/blob/development/src/user.ts#L415

**Comment by sadiabbasi** [comment]
> there is no new function `getDid()` on the user class. this needs to be added, you can copy most of the code from https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/blob/development/src/user.ts#L415
I didn't understand this point i didn't create any new function in user.ts i used this.getDid() so where i need to copy this code?

**Comment by theblockstalk** [comment]
```
getDid() return this.storage.did

createDid() should do what getDid() currently does, and be called in login() and createPerson()
```
The point of this is that the `storage.did` variable should be set when the user object is initialised, and then accessed later on with a getter function. What we currently do is that it is accessed later by creating it each time, instead of retrieving it from storage, This is less computationally efficient and not a very good class structure.

File: src/user.ts [hunk 3]
```diff
@@ -412,14 +419,15 @@ export class User {
...
-    async getDid() {
+    async createDid() {
         if (!this.chainID) {
             this.chainID = (await getChainInfo()).chain_id as unknown as Checksum256;
         }
 
         const accountName = await this.storage.accountName;
 
-        return `did:antelope:${this.chainID}:${accountName.toString()}`;
+        this.storage.did = `did:antelope:${this.chainID}:${accountName.toString()}`;
```
Comment (theblockstalk) [inline src/user.ts:429]:
```ts
this.storage.did = `did:antelope:${this.chainID}:${accountName.toString()}`;
await this.storage.did
return this.storage.did;
```

Comment (theblockstalk) [reply src/user.ts:429]: you need to await the storage object for it to persistently store the variable

File: src/user.ts [hunk 4]
```diff
@@ -412,14 +419,15 @@ export class User {
         return await Message.sign(payload, issuer, recipient);
     }
 
-    async getDid() {
+    async createDid() {
```
Comment (theblockstalk) [inline src/user.ts:422]: please add return type = string to this function.

also add JSDoc documentation
https://devhints.io/jsdoc

**Review by theblockstalk — COMMENTED** [review]
good work, a few final fixes

**Review by theblockstalk — APPROVED** [review]

**Pull request merged**